Move app settings into their own Vuex module

snowteamer opened the pull request:
Closes https://github.com/okTurtles/group-income/issues/1344
Closes https://github.com/okTurtles/group-income/issues/1276

### Summary of changes:

- Settings like appLogsFilter, fontSize, theme etc. have been moved to `@model/settings/vuexModule.js`.
- They should now correctly revert to their default values upon successfully logging out.
    If a `?debug` query parameter is present and truthy, then debug- and log-level applications logs will be enabled by default.
    Users can still disable them later via the Application Logs modal though.

snowteamer committed:
Move user settings to their own Vuex module

snowteamer committed:
Rename setAppLogsFilters to setAppLogsFilter

snowteamer committed:
Fix app settings not resetting on logout

snowteamer committed:
Enable debug logging by default if '?debug=true'

snowteamer committed:
Fix Flow errors

snowteamer committed:
Remove 'checked' attributes thanks to v-model

snowteamer committed:
Merge master into settings-vuex-module

snowteamer committed:
Merge branch 'master' into settings-vuex-module

snowteamer committed:
Run grunt pin:0.0.1

taoeffect reviewed (changes requested):
Great work @snowteamer!

I tested adjusting some settings, and while most seem to work, changing the system theme from Dark to Light and back while the settings are set to "System" doesn't work. I tried in both Firefox Developer Edition (with preferences set to "System") and also with Safari.

snowteamer committed:
Fix the Flow error

snowteamer committed:
Rename SelectorTheme to ThemeSelector

snowteamer committed:
Rename Theme.vue to ThemeSelector.vue

snowteamer committed:
Move @model/utils/themes.js to settings/themes.js

snowteamer committed:
Move @model/colors.js to settings/colors.js

snowteamer committed:
Remove unused colorManipulation.js import

snowteamer committed:
Fix system theme change event handler

corrideat commented on Gruntfile.js:
```diff
@@ -113,12 +113,12 @@ module.exports = (grunt) => {
     console.log(stdout)
     const r = /contracts\/([^.]+)\.(?:x|[\d.]+)\.manifest.*data\/(.*)/g
-    const manifests = Object.fromEntries(Array.from(stdout.matchAll(r), x => [`gi.contracts/${x[1]}`, x[2]]))
+    const manifests = Object.fromEntries(Array.from(stdout.replace(/\\/g, '/').matchAll(r), x => [`gi.contracts/${x[1]}`, x[2]]))
```
Is this some Windows-related change? Either `/` or `\\` should be valid FS path separators

snowteamer committed:
Merge master into settings-vuex-module

snowteamer committed:
Increase Cypress command timeout to 20s

snowteamer committed:
Restore LOGOUT handler in settings vuex module

taoeffect committed:
Archive proposals implemented (#1388)

* Closes #925 (Archive Proposals) + Closes #1377 (default threshold 66%)

+ Implemented archiving of proposals
+ Greatly improved security of contracts sandbox (almost complete)
+ Default percentage rule threshold is now 66%
+ Contracts can now have version-tracked SBP methods added to them via the "methods:" key

* updated pinned contracts at version 0.0.3

* @snowteamer review + additional fixes and improvements

snowteamer committed:
Add 'X-Frame-Options: deny' header to server responses (#1392)

* Add 'X-Frame-Options: deny' to every response

* Update avatar caching test

* Use chalk.yellow in warning

taoeffect commented:
It's surprising that this is failing repeatedly, even after `master` is merged.

@snowteamer what's your thinking about what might be the problem? Maybe it's a real bug in the PR?

pieer committed:
Fix wrapping box (#1397)

pieer committed:
Fix #1304:  Implement new design for automated ban proposals. (#1365)

* WIP update new proposition interface

* Refactor proposals

* Fix linting

* Fix test

* Fix proposal date

* Fix proposal messages and dates

* Add proposal modal

* Fix modal proposal

* Fix typo and remove console log

* Add selectbox

* Implement new design for automated ban proposals.

* Update automated ban proposal interface

* Fix  automated ban system

SebinSong committed:
Update payments page - task 1. , 2. , 3. #1339 (#1390)

* add distribution chip & banner to the payments list

* double quotes to single quotes

* style issue fix

* make some changes to PaymentsList.vue and its children

* update the other two tables / make the all-check checkbox work in payments-todo table

* update record-payments component

* work on PaymentsRowTodo and resolve cypress test issues

* add SendPaymentsLightning.vue Modal and create fake users

* resolve linter error

* add custom test-files for dev aid

* build PaymentRowSendLightning.vue component

* add toggle-btns for comment and the donation-fee

* add placeholder for qr-code sections in the modal

* remove trailing space

* build QrCode.vue generic component and add it to SendPaymentLightning.vue

* build CopyableInput.vue generic-component and add it to the Lightning payment modal

* resolve linter errors and fix the typo on LightningPayment modal

* work on mobile&tablet of TODO table

* work on mobile&tablet screen designs for multiple payments-related components

* work on mobile design of lightning record item

* additional work to PaymentPagination

* remove temp cypress test

* work on gerg's CR ro .1

* fix a bug on lightning modal

* remove trailing space

* fix some test bug

* work on Pierre's CR

* work on Greg's CR Ro.2 - implement 'relative to' in the table

* fix for next distribution date pill

* fix the duplicated distribution pill

* fix the pill bug / add anchor tag for distribution info

* fix some design issues

* fix failing cypress test items

SebinSong committed:
imporve 'grunt pin' to make it run esbuild first (#1402)

pieer committed:
Fix chat not working on mobile (#1326)

* Fix text area qnd button

* Update send button

* Fix linting

* Fix send box

SebinSong committed:
#1339 updated payments page - task [4] (#1406)

* add distribution chip & banner to the payments list

* double quotes to single quotes

* style issue fix

* make some changes to PaymentsList.vue and its children

* update the other two tables / make the all-check checkbox work in payments-todo table

* update record-payments component

* work on PaymentsRowTodo and resolve cypress test issues

* add SendPaymentsLightning.vue Modal and create fake users

* resolve linter error

* add custom test-files for dev aid

* build PaymentRowSendLightning.vue component

* add toggle-btns for comment and the donation-fee

* add placeholder for qr-code sections in the modal

* remove trailing space

* build QrCode.vue generic component and add it to SendPaymentLightning.vue

* build CopyableInput.vue generic-component and add it to the Lightning payment modal

* resolve linter errors and fix the typo on LightningPayment modal

* work on mobile&tablet of TODO table

* work on mobile&tablet screen designs for multiple payments-related components

* work on mobile design of lightning record item

* additional work to PaymentPagination

* work on gerg's CR ro .1

* fix a bug on lightning modal

* work on Pierre's CR

* work on Greg's CR Ro.2 - implement 'relative to' in the table

* fix for next distribution date pill

* fix the duplicated distribution pill

* add btn for lightning payment detail modal

* fix the pill bug / add anchor tag for distribution info

* add LinkToCopy to the payment detail modal for lightning

* fix the padding too much bug

* centralize the lightning dummy data

* remove the unused test file

snowteamer committed:
Cache namespace lookups in Vuex state (#1409)

* Cache namespace lookups in Vuex

* Apply review feedback (@taoeffect)

SebinSong committed:
Bug fixes on payments page / Resolving issue #1367 (#1410)

* fix the issue #1367 user-settings menu is hidden on mobile

* resolve a couple of minor UI bugs in Payments page

taoeffect committed:
update chel to v1.1.3 (#1417)

Silver-IT committed:
Improve Test Scenario & Fix UI Issues in Group Chat (#1341)

* feat: added init test case for group chat scrolling

* faet: reply message and fix popup and tooltip visibility after click

* fix: click reply button issue when ChatMain page is scrollable

* feat: reply message and scrolling to the target

* feat: improve test scenario for group chat scrolling

* feat: check infinite scroll & saved scroll position

* fix: error regarding the class name

* fix: absolute position of messages while reply message and scroll up

* feat: update chat pannel height considering new height of textarea & fix: cypress erorr

* feat: updated target message when it is removed

* feat: added test scenario for mentioning and notifications and fixed error in making mentioning

* feat: grunt pin:0.0.4

* feat: improved test scenario to check mentions

* fix: UI issue when editing message

* fix: prettier issue

* fat: make notification even though identity contract is not synced

* chore: removed unnecessarity

* feat: DRY for the computed values

* feat: updated translation

* feat: reverted contract version 0.0.4

* chore: reverted contract version 0.0.4

SebinSong committed:
fix the bug where side-bar is cropped at the bottom (#1414)

pieer committed:
Dashboard tab Skeleton (#1404)

* Add different dashboard tabs

* Add more activities panels

* Add inacitve users panel

* Cleaning

* Fix greg PR

SebinSong committed:
Remove disagreement rule from group creation step (#1407)

* remove disagreement rules UI/Logic from the group creation step

* change for greg's CR

* more update to the UI of voting rule

* update to the design of the voting rule step

* remove disagreement rule from the group-settings page

* revert back to the original code in GroupRulesSettings.vue

* remove 'active' tag from the UI

* copy update

snowteamer committed:
Add LRU cache to the backend (#1424)

* Add lru-cache.js to the backend

* Use node-lru-cache instead of custom cache

* Apply @taoeffect review

SebinSong committed:
#1415 - Add 'confirm password' field to the sign-up form (#1422)

* add confirm password field to SignUpForm.vue

* writing a cypress test for new signup form

* improve the validation form logic in PasswordForm.vue

* random change for flaky cypress on github

* right padding for when there is a single addon

* fix the bug where the text-color is diff when errored

SebinSong committed:
#1339 Update payments page - subtask 5. (#1421)

* add distribution chip & banner to the payments list

* double quotes to single quotes

* style issue fix

* make some changes to PaymentsList.vue and its children

* update the other two tables / make the all-check checkbox work in payments-todo table

* update record-payments component

* work on PaymentsRowTodo and resolve cypress test issues

* add SendPaymentsLightning.vue Modal and create fake users

* resolve linter error

* add custom test-files for dev aid

* build PaymentRowSendLightning.vue component

* add toggle-btns for comment and the donation-fee

* add placeholder for qr-code sections in the modal

* remove trailing space

* build QrCode.vue generic component and add it to SendPaymentLightning.vue

* build CopyableInput.vue generic-component and add it to the Lightning payment modal

* resolve linter errors and fix the typo on LightningPayment modal

* work on mobile&tablet of TODO table

* work on mobile&tablet screen designs for multiple payments-related components

* work on mobile design of lightning record item

* additional work to PaymentPagination

* work on gerg's CR ro .1

* fix a bug on lightning modal

* work on Pierre's CR

* work on Greg's CR Ro.2 - implement 'relative to' in the table

* fix for next distribution date pill

* fix the duplicated distribution pill

* add btn for lightning payment detail modal

* fix the pill bug / add anchor tag for distribution info

* add LinkToCopy to the payment detail modal for lightning

* fix the padding too much bug

* centralize the lightning dummy data

* work on re-arranging the not-received badges

* create Modals for Thank You Note

* small UI fix in /payments page

* add field for paymentThankYou to group state

* hook the actual thank you note state to the UI elements

* add cypress test for task 5.

* delete the dev-only cypress test file

* use alert() instead of console.error()

* work on Greg's CR ro.1

* bump contracts to v0.0.6

* work on Leihla's CR

* work on Greg's CR ro.2

* update contract v0.0.6 again

* update the PAYMENT_RECEIVED notification logic

* fix for the cypress test item

* update contracts v0.0.6 to reflect the recent changes

* changes for CR from Alex

* update contracts v0.0.6 again

* update for flaky cypress test

* explicitly call replaceModal() if another modal is queried when one is already open

* fix the typos

Silver-IT committed:
Expiring proposal notification (#1423)

* feat: expiring proposal notification

* feat: yellow horn when the proposal expiring

* fix: typo & improved codes using utility functions

* fix: interactive messages from proposals

* feat: remove contracts 0.0.6

* feat: resolved conflicts

* feat: translation typo

* feat: message for generic proposals

* chore: for cypress retry

* chore: cypress retry

* feat: ignore ChelErrorUnexpected

* feat: improve translations

SebinSong committed:
remove disagreeing number option from the proposal dropdown (#1437)

snowteamer committed:
No longer ignore @model/contracts/ in .flowconfig (#1444)

SebinSong committed:
#1426 GroupCreationModal reachable even when no group is created (#1445)

* make sure the group creation steps are not reachable by manual route change

* move the pre-auth gaurd to router.js

* remove unused getter

Silver-IT committed:
Resolve frontend issues (#1441)

* feat: reactive group settings form and avatar to state

* feat: added postEffect in submit function of SignUp and Login forms

* feat: improve prop name

* fix: validation issues regarding chatroom name and description

* feat: add comments for better understanding

SebinSong committed:
#658 Add a profile picture editor (#1442)

* create a modal and scaffolds for various UIs in it

* sync <slider-continuous /> with +/- buttons / extract <canvas /> into a standalone component / initial setup of canvas upon image load

* implement initial draw of the image on canvas once the image is loaded

* fix various bugs / apply zoom to the painted image

* implement image crop / extract / turning into dataURL / event emittion for it

* add avatar editor to the group-settings page / update the cypress test

* further update to the cypress test

* place various contants in one central place

* apply css filters to make the clipping circle noticeable

* output image as a square jpeg

* make the generated image as a square jpeg

* additional test update to signup-and-login.js

* add mouse-wheel event handler to the editor-canvas

* cursor back to the default

* add translation action to the canvas

* add translation action to the canvas

* setup pinch gesture on the canvas

* instruction copy change depending on whether touchable device or not

* work on Greg's CR for devtool issue

* more cancel handler / remove unused local state

* limit the translation of the image

Silver-IT committed:
Support History Widget Implemented (#1434)

* fix: proposal widget title style

* feat: support history percentage implementation in dashboard

* feat: grunt pin:0.0.8

* feat: grunt pin:0.0.8

* feat: archive payments

* feat: remove DRY and improve codes and naming variables

* fix: display MAX_HISTORY_PERIODS of support history

* feat: get historical payments for support history

* feat: grunt pin

* feat: pagination of payments and periods

* chore: removed unnecessary getter

* feat: improve the meaning of constant and to archive payments for upcoming features to change constant

* feat: removed functions in contract shared

* feat: improve code splitting functions into details

* feat: payments pagination

* feat: improve code and set support history percentage 0 when no payments

* fix: call group getters function only when state is vuex

* feat: improve performance regarding currentPaymentPeriod

* feat: improve the performance of reactiveDate

SebinSong committed:
make the payment method unselectable (#1452)

snowteamer committed:
Fix merge conficts
